### ISSUE TITLE
authorization_rules/change line(35) undefined method `to_sym' for nil:NilClass

### ISSUE THREAD

**Issue by bublik** [stffn/declarative_authorization#15]
Showing /usr/lib/ruby-enterprise-1.8.7/lib/ruby/gems/1.8/gems/declarative_authorization-0.4/app/views/authorization_rules/_change.erb where line #35 raised:

undefined method `to_sym' for nil:NilClass

Extracted source (around line #35):

32:   </thead>
33:   <tbody>
34:     <% @users.each do |user| %>
35:       <tr class="<%= controller.authorization_engine.permit?(@privilege, :context => @context, :user => user, :skip_attribute_test => true) ? 'permitted' : 'not-permitted' %>">
36:         <td class="user_id"><%=h user.id %></td>

**Comment by zacheryph** [comment]
can you print a backtrace for this please.

**Comment by bublik** [comment]
/usr/lib/ruby-enterprise-1.8.7/lib/ruby/gems/1.8/gems/activesupport-2.3.5/lib/active_support/whiny_nil.rb:52:in `method_missing'
/usr/lib/ruby-enterprise-1.8.7/lib/ruby/gems/1.8/gems/declarative_authorization-0.4/lib/declarative_authorization/authorization.rb:142:in`permit!'
/usr/lib/ruby-enterprise-1.8.7/lib/ruby/gems/1.8/gems/declarative_authorization-0.4/lib/declarative_authorization/authorization.rb:184:in `permit?'
/usr/lib/ruby-enterprise-1.8.7/lib/ruby/gems/1.8/gems/declarative_authorization-0.4/app/views/authorization_rules/_change.erb:35:in`_run_erb_47usr47lib47ruby45enterprise45146846747lib47ruby47gems47146847gems47declarative_authorization45046447app47views47authorization_rules47_change46erb_locals_change_object'
/usr/lib/ruby-enterprise-1.8.7/lib/ruby/gems/1.8/gems/declarative_authorization-0.4/app/views/authorization_rules/_change.erb:34:in `each'
/usr/lib/ruby-enterprise-1.8.7/lib/ruby/gems/1.8/gems/declarative_authorization-0.4/app/views/authorization_rules/_change.erb:34:in`_run_erb_47usr47lib47ruby45enterprise45146846747lib47ruby47gems47146847gems47declarative_authorization45046447app47views47authorization_rules47_change46erb_locals_change_object'
/usr/lib/ruby-enterprise-1.8.7/lib/ruby/gems/1.8/gems/actionpack-2.3.5/lib/action_view/renderable.rb:34:in `send'
/usr/lib/ruby-enterprise-1.8.7/lib/ruby/gems/1.8/gems/actionpack-2.3.5/lib/action_view/renderable.rb:34:in`render'
/usr/lib/ruby-enterprise-1.8.7/lib/ruby/gems/1.8/gems/actionpack-2.3.5/lib/action_view/base.rb:306:in `with_template'
/usr/lib/ruby-enterprise-1.8.7/lib/ruby/gems/1.8/gems/actionpack-2.3.5/lib/action_view/renderable.rb:30:in`render'
/usr/lib/ruby-enterprise-1.8.7/lib/ruby/gems/1.8/gems/actionpack-2.3.5/lib/action_view/renderable_partial.rb:20:in `render'
/usr/lib/ruby-enterprise-1.8.7/lib/ruby/gems/1.8/gems/actionpack-2.3.5/lib/action_controller/benchmarking.rb:26:in`benchmark'
/usr/lib/ruby-enterprise-1.8.7/lib/ruby/gems/1.8/gems/activesupport-2.3.5/lib/active_support/core_ext/benchmark.rb:17:in `ms'

**Comment by zacheryph** [comment]
do you possible have any simple instructions for reproducing this. or as easily a project to create and hub it up or gist it

**Comment by bublik** [comment]
Crete users with privilegies
   ['admin', 'moderator', 'user'].each do |priv|
      u = User.create(:login => priv, :name => "User #{priv}", :email => "#{priv}@#{APP_CONFIG['domain']}", :password => priv, :password_confirm => priv)
      u.activate!
      u.roles.create(:name => priv)

##     end

User model 
  has_many :videos
  has_many :assignments
  has_many :roles, :through => :assignments

  def role_symbols
    (roles || []).map {|r| r.name.underscore.to_sym}

##   end

Next step open 
http://localhost:3000/authorization_rules/change
And have error 

undefined method `to_sym' for nil:NilClass
In line 
 <tr class="<%= controller.authorization_engine.permit?(@privilege, :context => @context, :user => user, :skip_attribute_test => true) ? 'permitted' : 'not-permitted' %>">

---

My authorization rules 

authorization do
  role :guest do
    has_permission_on [:videos, :thumbnails, :categories], :to => [:index, :show, :search, :tag]
    has_permission_on [:users], :to => [:create, :new, :activate]
    #    has_permission_on :comments, :to => [:new, :create]
    #    has_permission_on :comments, :to => [:edit, :update] do
    #      if_attribute :user => is { user }
    #    end
  end

  role :user do
    includes :guest
    has_permission_on :videos, :to => [:edit, :update] do
      if_attribute :user => is { user }
    end
  end

  role :moderator do
    includes :guest
    includes :user
    has_permission_on [:videos, :categories], :to => [:edit, :update]
  end

  role :admin do
    includes :moderator
    has_permission_on [:authorization_rules, :authorization_usages], :to => :read
    has_permission_on [:videos, :categories], :to => [:new, :create,  :destroy, :delete]
  end

end

**Comment by bublik** [comment]
Back trace 

```
/usr/lib/ruby/gems/1.8/gems/declarative_authorization-0.4/app/controllers/./../lib/declarative_authorization/authorization.rb:142:in `permit!'
/usr/lib/ruby/gems/1.8/gems/declarative_authorization-0.4/app/controllers/./../lib/declarative_authorization/authorization.rb:184:in `permit?'
/usr/lib/ruby/gems/1.8/gems/declarative_authorization-0.4/app/views/authorization_rules/_change.erb:35
/usr/lib/ruby/gems/1.8/gems/declarative_authorization-0.4/app/views/authorization_rules/_change.erb:34:in `each'
/usr/lib/ruby/gems/1.8/gems/declarative_authorization-0.4/app/views/authorization_rules/_change.erb:34
/usr/lib/ruby/gems/1.8/gems/declarative_authorization-0.4/app/views/authorization_rules/change.html.erb:168
/usr/lib/ruby/gems/1.8/gems/declarative_authorization-0.4/app/controllers/authorization_rules_controller.rb:47:in `change'
```

**Comment by stffn** [comment]
You don't seem to have any rules in your project's authorization rules.  Still, this shouldn't result in an exception.  I will fix that.

**Comment by stffn** [comment]
Oh, actually, it is about not having any privilege hierarchies in the rules file.

**Comment by stffn** [comment]
Fixes exception in change support for rules file without privilege hierarchy (Closed by aac2336cdd9f6c27dbcfa4f49ad4ed5dafbcc19d)
